src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
+  const reorderedDate = [];
+  const oldYear = newDate[oldYearIndex];
+
+  function getIndex(index) {
+    return index.indexOf('YY') !== -1
+      ? index.indexOf('YY')
+      : index.indexOf('YYYY');
+  }
```

Comment:
In the getIndex function, you are using the indexOf method twice for the 'YY' string. This is unnecessary and can be optimized by storing the result in a variable and then using it in the condition.